cluster/gce/upgrade.sh

```diff
@@ -439,6 +439,17 @@ function do-node-upgrade() {
   echo "== Finished upgrading nodes to ${KUBE_VERSION}. ==" >&2
 }
 
+function update-coredns-config() {
+  # Deletes the existing CoreDNS ConfigMap so that the latest can be applied.
+  echo "== Deleting existing CoreDNS ConfigMap. ==" >&2
+  "${KUBE_ROOT}/cluster/kubectl.sh" delete configmap coredns --namespace=kube-system 2>/dev/null
```

Comment:
isn't this data loss? I thought that configmap contained user data

Reply:
Yep this may contains user data. We were under the impression that this script is not used in production in any way. Otherwise we will need something like https://github.com/kubernetes/kubernetes/pull/78033 in the script. Or maybe do what was recommended on https://github.com/kubernetes/kubernetes/pull/78033#issuecomment-498328908.

Reply:
oh. I have no idea what calls this script, will defer to cluster/gce owners

Reply:
> We were under the impression that this script is not used in production in any way

if that turns out to be true, and we do go forward with this change, there needs to be a comment here documenting that this is known to delete a configmap that contains user data, and that it is ok because this script is not used by any production setups. I could easily see this being copied-pasted as "the way" to clear/upgrade the dns configmap

Reply:
@liggitt I have added comments to warn that this shouldn't be followed in a production setup. Let me know if this is acceptable.

We still need to verify whether this script is used in production or not.

Reply:
@bowei, is `cluster/gce/upgrade.sh` prescribed for use in production clusters in any way?

Reply:
Searching the kubernetes repo, I only see use of this script from within the e2e/tests.

However, I found references in kubernetes/website ... in this page ...

https://kubernetes.io/docs/tasks/administer-cluster/cluster-management/

> Upgrades on open source Google Compute Engine (GCE) clusters are controlled by the cluster/gce/upgrade.sh script.
> Get its usage by running cluster/gce/upgrade.sh -h.

So, apparently we _do_ prescribe its usage in production.

Which means we cannot merge this as is, which will tumble a few more dominoes...